addons/analytic/models/analytic_account.py

```diff
@@ -82,16 +82,16 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
     def _compute_debit_credit_balance(self):
         Curr = self.env['res.currency']
         analytic_line_obj = self.env['account.analytic.line']
-        domain = [('account_id', 'in', self.ids)]
+        domain = [
+            ('account_id', 'in', self.ids),
+            ('company_id', 'in', self.company_id.ids or self.env.companies.ids)]
```

Comment:
This is incorrect. You are working with a recordset... In fact this compute is clearly shitty and is an open door to multi-company issues. If we make the assumption this is called only from the analytic report, I think `self.env.companies.ids` is enough.

Reply:
With a recordset, self.company_id.ids works also since some orm changes some months ago.
I'm nearly sure it crashed when i only used self.env.companies.ids, but I can try again.